### PR TITLE
3217 - Fix enter all digits for fr-fr with locale

### LINKED ISSUE #3217: Locale: unable to enter all digits for a decimal number when locale is fr-FR
**Describe the bug**
Unable to enter all digits for a decimal number when the locale is set to fr-FR.

**To Reproduce**
1. Go to http://latest-enterprise.demo.design.infor.com/components/mask/test-number-mask-gauntlet.html?locale=fr-FR
2. For the Number: 1 000,00 field (column 1, second row), try to enter 1234,
3. The error is when the decimal separator is entered, the value becomes 123,
4. The expectation is the value is 1 234,
5. For the Millions: 1 000 000,00 field (column 1, fourth row), try to enter 1234567,
6. The error is the last digit of 7 cannot be entered, the value becomes 123 456
7. The expectation is the value is 1 234 567,
8. For the Billions: 1 000 000 000,00 field (column 1, sixth row), try to enter 1234567890,
9. The error is the last digit that can be entered is 8, the value becomes 12 345 678
10. The expectation is the value is 1 234 567 890,

**Expected behavior**
Expectations are stated above.

**Version**

- ids-enterprise: 4.23.0

**Screenshots**
locale-number-mask-fr-FR-issue

**Platform**
 - All

**Additional context**
For the fr-FR locale, also seeing NaN in the placeholder and labels for the Quintillion fields.

### PR DESCRIPTION
**Pull request by deep7102**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**
Fixed enter all digits was not working for fr-FR with Locale.

**Related github/jira issue (required)**:
Closes #3217

**Steps necessary to review your pull request (required)**:
- Pull this branch and build/run the demo app
- Navigate to http://localhost:4000/components/mask/test-number-mask-gauntlet.html?locale=fr-FR
- For the Number: 1 000,00 field, enter `1234,`
- Should let you enter and the value should change to `1 234,`
- For the Millions: 1 000 000,00 field, enter 1234567,
- Should let you enter and the value should change to `1 234 567,`
- For the Billions: 1 000 000 000,00 field, enter 1234567890,
- Should let you enter and the value should change to `1 234 567 890,`
- Also see the Quintillion fields should not be any NaN in the placeholder and labels

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

### CODE REVIEW

File: test/components/mask/mask.e2e-spec.js [hunk 1]
```diff
@@ -45,4 +45,37 @@ describe('Mask Percent Format Tests', () => {
 
     expect(await inputEl.getAttribute('value')).toEqual('100 ٪');
   });
+
+  it('Should be type in fr-FR', async () => {
```
Comment (tmcconechy) [inline test/components/mask/mask.e2e-spec.js]:
```suggestion
  it('Should be able to type in fr-FR', async () => {
```

File: src/components/locale/locale.js [hunk 2]
```diff
@@ -815,13 +815,34 @@ const Locale = {  // eslint-disable-line
...
+    return n;
+  },
+
+  /**
+  * Find browser default separator for given locale
+  * @param {string} locale The locale
+  * @param {string} separatorType The separator type be found `group`|`decimal`
+  * @returns {string} The browser default separator character
+  */
+  getSeparator(locale, separatorType) {
```
Comment (tmcconechy) [inline src/components/locale/locale.js]: Should this be private? Or is it useful?

Comment (deep7102) [reply src/components/locale/locale.js]: Good catch, should be private

**Comment by tmcconechy** [comment]
@EdwardCoyle whats your opinion on my comment https://github.com/infor-design/enterprise/pull/3314#pullrequestreview-341868854 ?

**Comment by EdwardCoyle** [comment]
@tmcconechy It seemed like the mask was working as intended to me -- I thought `1 234,` was the proper formatting for that mask on `fr-FR`.

**Comment by tmcconechy** [comment]
Yep i think i made a mistake testing. It was like that.

**Pull request merged**